core/lib/zksync_core/src/sync_layer/fetcher.rs

```diff
@@ -26,6 +26,11 @@ pub struct MainNodeFetcherCursor {
     pub(super) l1_batch: L1BatchNumber,
 }
 
+struct FetcherIterationResult {
```

Comment:
Nit: Please use `#[derive(Debug)]` for all types (even private ones); it simplifies debugging, logging etc.